library/core/src/main/java/com/google/android/exoplayer2/extractor/ts/H262Reader.java

```diff
@@ -130,7 +148,25 @@ public void consume(ParsableByteArray data) {
           hasOutputFormat = true;
         }
       }
+      if (userDataReader != null && userData != null) {
+        int lengthToStartCode = startCodeOffset - offset;
+        int bytesAlreadyPassed = 0;
+        if (lengthToStartCode > 0) {
+          userData.appendToNalUnit(dataArray, offset, startCodeOffset);
+        } else {
+          bytesAlreadyPassed = -lengthToStartCode;
+        }
+
+        if (userData.endNalUnit(bytesAlreadyPassed)) {
+          //send media format stuff here.
+          userDataParsable.reset(userData.nalData, userData.nalLength);
+          userDataReader.consume(sampleTimeUs, userDataParsable);
+        }
 
+        if (startCodeValue == START_USER_DATA && data.data[startCodeOffset + 2] == 0x1) {
```

Comment:
I think you can do `userData.startNalUnit(startCodeValue)` unconditionally. `startNalUnit` already checks the type internally, and I think the data.data check is guaranteed to be true due to `NalUnitUtil.findNalUnit` having found a NAL unit further up in this method.

Reply:
We need to check two bytes : 0x01B2.